common/lib/capa/capa/tests/test_inputtypes.py

```diff
@@ -781,25 +779,30 @@ def test_get_html(self):
             """).replace('\n', '')
 
         output = self.the_input.get_html()
-        if six.PY2:
-            expected_string = """
-            <div class='matlabResponse'><div id='mwAudioPlaceHolder'>
-            <audio src='data:audio/wav;base64=' autobuffer="" controls="" autoplay="">Audio is not supported on this browser.</audio>
-            <div>Right click <a href="https://endpoint.mss-mathworks.com/media/filename.wav">here</a> and click \"Save As\" to download the file</div></div>
-            <div style='white-space:pre' class='commandWindowOutput'/><ul/></div>
-            """
-        else:
-            expected_string = """
-            <div class="matlabResponse"><div id="mwAudioPlaceHolder">
-            <audio autobuffer="" controls="" autoplay="" src="data:audio/wav;base64=">Audio is not supported on this browser.</audio>
-            <div>Right click <a href="https://endpoint.mss-mathworks.com/media/filename.wav">here</a> and click \"Save As\" to download the file</div></div>
-            <div style="white-space:pre" class="commandWindowOutput"/><ul/></div>
-            """
-        self.assertEqual(
-            etree.tostring(output).decode('utf-8'),
-            textwrap.dedent(expected_string).replace('\n', '').replace('\'', '\"')
-        )
-
+        elements = []
+        element_tags = []
+        element_keys = []
+        for element in output.iter():
+            elements.append(element)
+            element_tags.append(element.tag)
+            element_keys.append(element.keys())
+        assert element_tags.count('div') == 4
+        assert element_tags.count('audio') == 1
+        audio_index = element_tags.index('audio')
+
+        six.assertCountEqual(self, element_keys[audio_index], ['autobuffer', 'controls', 'autoplay', 'src'])
+        self.assertEqual(elements[audio_index].get('src'), 'data:audio/wav;base64=')
+        self.assertEqual(elements[audio_index].text, 'Audio is not supported on this browser.')
+        href_index = element_keys.index(['href'])
+        self.assertEqual(elements[href_index].get('href'), 'https://endpoint.mss-mathworks.com/media/filename.wav')
+        id_index = element_keys.index(['id'])
+        self.assertEqual(elements[id_index].get('id'), 'mwAudioPlaceHolder')
+        output_string = etree.tostring(output).decode('utf-8')
+
+        six.assertRegex(self, output_string, "div.*mwAudioPlaceHolder")
+        six.assertRegex(self, output_string, "audio.*Audio is not supported.*audio")
+        six.assertRegex(self, output_string, "div.*Right click.*href.*media.*here.*click.*download.*div")
+        six.assertRegex(self, output_string, "div .*style=.*ul.*div")
```

Comment:
Maybe use equivalent assertions on the attributes and element text of the etree hierarchy instead of converting the whole thing to text and doing regexes on that?